homeassistant/components/fritz/__init__.py

```diff
@@ -1 +1,90 @@
-"""The fritz component."""
+"""Support for AVM Fritz!Box functions."""
+import logging
+
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import (
+    CONF_DEVICES,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .common import FritzBoxTools
+from .const import (
+    DATA_FRITZ_TOOLS_INSTANCE,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DOMAIN,
+    ERROR_CONNECTION_ERROR,
+    SUPPORTED_DOMAINS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up FRITZ!Box Tools component."""
+    if DOMAIN in config:
+        for entry_config in config[DOMAIN][CONF_DEVICES]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=entry_config
+                )
+            )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up fritzboxtools from config entry."""
+    _LOGGER.debug("Setting up FRITZ!Box Tools component")
+    host = entry.data.get(CONF_HOST, DEFAULT_HOST)
+    port = entry.data.get(CONF_PORT, DEFAULT_PORT)
+    username = entry.data.get(CONF_USERNAME)
+    password = entry.data.get(CONF_PASSWORD)
+
+    fritz_tools = FritzBoxTools(
+        hass=hass,
+        host=host,
+        port=port,
+        username=username,
+        password=password,
+    )
+
+    success, error = await hass.async_add_executor_job(
+        lambda: fritz_tools.async_setup()
+    )
+
+    if not success and error is ERROR_CONNECTION_ERROR:
+        _LOGGER.error("Unable to setup FRITZ!Box Tools component")
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_REAUTH},
+                data=entry,
+            )
+        )
+        return False
+
+    hass.data.setdefault(DOMAIN, {DATA_FRITZ_TOOLS_INSTANCE: {}, CONF_DEVICES: set()})
```

Comment:
Are we sharing data between config entries ?

Can this all be under `hass.data[DOMAIN][entry.entry_id]` ?

Reply:
```suggestion
    hass.data.setdefault(DOMAIN, {})
```

Reply:
Hi,
in further PRs we might want to merge the services [`reboot`](https://github.com/mammuth/ha-fritzbox-tools/blob/04836d540a8e1f42cd4d2d552191cf2c0a5bc6f4/custom_components/fritzbox_tools/__init__.py#L112) and `reconnect` which are currently not connected to the entries. 
Is there any need/advantage to change the structure in `hass.data`